aredis_om/model/model.py

```diff
@@ -742,8 +742,12 @@ async def execute(self, exhaust_results=True):
             # current offset plus `page_size`, until we stop getting results back.
             query = query.copy(offset=query.offset + query.page_size)
             _results = await query.execute(exhaust_results=False)
-            if not _results:
-                break
+            if exhaust_results:
```

Comment:
Can these two statements combine?
```if exhaust_results and not _results:
    break
````

Reply:
Doesn't seem important enough to block merging this. @chayim Can you approve?
Or would you like me to create a new PR?